### PR TITLE
rfr - 8 Add Settings > Channel page

### PR DESCRIPTION
**Pull request by Andruhin**
Added Channels list and channels details activities. Implemented functionality for adding/saving/removing/editing channels. Branch is instead of the previous rfr-16

### CODE REVIEW

File: app/src/main/java/rss/feed/reader/managers/ChannelManager.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+/**
+ * Created by Andriy Ksenych on 17.10.2016.
+ * Manager for saving/managing user input channels list
+ */
+
+public class ChannelManager {
+
+    private static final String PREF_CHANNELS_LIST = "channels list";
+
+    public Gson mGson;
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/managers/ChannelManager.java]: why public ?

File: app/src/main/java/rss/feed/reader/managers/ChannelManager.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        mPreferences.edit().putString(PREF_CHANNELS_LIST, getJsonStringFromList(list)).commit();
+    }
+
+    private List<Channel> readChannelsList() {
+        return getChannelsListFromString(mPreferences.getString(PREF_CHANNELS_LIST, "[]"));
+    }
+
+    public List<Channel> getChannelsList() {
+        if (mChannelsList == null) {
+            mChannelsList = readChannelsList();
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/managers/ChannelManager.java]: don't we have a bug here? If you invoke getChannelsList() first, then saveChannelsList() and then getChannelsList() will return previous list..

File: app/src/main/java/rss/feed/reader/managers/ChannelManager.java (hunk 1 above)
Comment (Andruhin) [inline app/src/main/java/rss/feed/reader/managers/ChannelManager.java]: forgot to change)

File: app/src/main/java/rss/feed/reader/managers/ChannelManager.java (hunk 2 above)
Comment (Andruhin) [inline app/src/main/java/rss/feed/reader/managers/ChannelManager.java]: saveChannelsList(...) should be private method and cann't be called from outside.

**Pull request merged**